sudo/src/run_handler.rs

```diff
@@ -169,15 +169,19 @@ fn adjust_args_for_gui_exes(req: &mut ElevateRequest) {
     tracing::trace_log_message(&format!("is_exe: {is_exe}"));
     tracing::trace_log_message(&format!("is_gui: {is_gui}"));
 
-    // TODO: figure out how to handle non-exe files. ShellExecute(runas,
-    // ...) doesn't do anything for them, and I'm not sure we can trivially
-    // have the service find out what the right verb is for an arbitrary
-    // extension. (this is the kind of comment I'm sure to be proven wrong
-    // about)
     if is_gui {
         tracing::trace_log_message("not cli exe. Force new window");
         req.sudo_mode = SudoMode::ForceNewWindow;
     }
+
+    // We're changing the argument to the application here and using the
+    // registry to figure out the default application for the specified
+    // file.
+    if !is_exe {
+        req.args
+            .splice(0..0, [req.application.clone()]);
+        req.application = get_default_application(&req.application).expect("Failed to get the default application");
```

Comment:
this might fail if a file doesn't have a file association with it. the standard behavior (non-sudo) is to open the open with prompt, so this might have to be changed before being merged